// .ignored_openclaw/dist/web-BCGH1b-T.js
import "./paths-tuenh9TL.js";
import "./subsystem-G54saDcg.js";
import "./utils-B8zCe27d.js";
import "./reply-cpXZjA1O.js";
import "./agent-scope-DmTcOjk4.js";
import "./openclaw-root-B5jXk2td.js";
import "./logger-BohkHL0S.js";
import "./exec-CNtAq7Md.js";
import "./github-copilot-token-Ff890Vmh.js";
import "./boolean-B938tROv.js";
import "./env-CQxWwTMn.js";
import "./env-overrides-BYyMsIvB.js";
import "./registry-aHEgjCIQ.js";
import "./skills-BtMFdOkk.js";
import "./frontmatter-BngyYhS0.js";
import { C as WA_WEB_AUTH_DIR, O as pickWebChannel, P as webAuthExists, T as logWebSelfId } from "./plugins-om7vnHk0.js";
import "./query-expansion-BNb6Vys6.js";
import "./redact-owRpqMhM.js";
import "./path-alias-guards-BEqdGdn9.js";
import "./fetch-CGW2R_wh.js";
import "./errors-DNKaAVHN.js";
import "./cmd-argv-BdNFwuhe.js";
import "./restart-stale-pids-ZfbSMILU.js";
import "./delivery-queue-B7zSt4zY.js";
import "./paths-CfmKJcO5.js";
import "./session-cost-usage-C2jrTKxW.js";
import "./prompt-style-DlcWUCNR.js";
import "./links-BpKafJs4.js";
import "./cli-utils-BSURvrmb.js";
import { n as monitorWebInbox, t as monitorWebChannel } from "./channel-web-B_hlprb5.js";
import { t as sendMessageWhatsApp } from "./send-1jaXrPxo.js";
import { i as waitForWaConnection, t as createWaSocket } from "./session-jaYDDyZR.js";
import { t as loginWeb } from "./login-DxiBcrRk.js";
export { WA_WEB_AUTH_DIR, createWaSocket, logWebSelfId, loginWeb, monitorWebChannel, monitorWebInbox, pickWebChannel, sendMessageWhatsApp, waitForWaConnection, webAuthExists };
