// .ignored_openclaw/dist/plugin-sdk/telegram-B4ZoFrqL.js
import { h as DEFAULT_ACCOUNT_ID } from "./session-key-CbP51u9x.js";
import { R as formatCliCommand } from "./channel-config-helpers-CX9AfjCJ.js";
import { E as formatDocsLink, do as resolveDefaultTelegramAccountId, fo as resolveTelegramAccount, lo as inspectTelegramAccount, uo as listTelegramAccountIds, vr as normalizeTelegramLookupTarget, yr as parseTelegramTarget } from "./thread-bindings-DxarhdX6.js";
import { c as isRecord } from "./utils-BhiRkVxe.js";
import { tt as hasConfiguredSecretInput } from "./zod-schema.core-GWl_s9xp.js";
import { T as splitOnboardingEntries, g as resolveOnboardingAccountId, h as resolveAccountIdForConfigure, l as patchChannelConfigForAccount, m as promptSingleChannelSecretInput, n as applySingleTokenPromptResult, p as promptResolvedAllowFrom, x as setOnboardingChannelEnabled, y as setChannelDmPolicyWithAllowFrom } from "./helpers-DAMRJRRS.js";
import { i as resolveEnabledConfiguredAccountId, n as asString, t as appendMatchMetadata } from "./shared-B4PQjsPw.js";
//#region extensions/telegram/src/api-fetch.ts
async function fetchTelegramChatId(params) {
	const url = `https://api.telegram.org/bot${params.token}/getChat?chat_id=${encodeURIComponent(params.chatId)}`;
	try {
		const res = await fetch(url, params.signal ? { signal: params.signal } : void 0);
		if (!res.ok) return null;
		const data = await res.json().catch(() => null);
		const id = data?.ok ? data?.result?.id : void 0;
		if (typeof id === "number" || typeof id === "string") return String(id);
		return null;
	} catch {
		return null;
	}
}
//#endregion
//#region extensions/telegram/src/onboarding.ts
const channel = "telegram";
async function noteTelegramTokenHelp(prompter) {
	await prompter.note([
		"1) Open Telegram and chat with @BotFather",
		"2) Run /newbot (or /mybots)",
		"3) Copy the token (looks like 123456:ABC...)",
		"Tip: you can also set TELEGRAM_BOT_TOKEN in your env.",
		`Docs: ${formatDocsLink("/telegram")}`,
		"Website: https://openclaw.ai"
	].join("\n"), "Telegram bot token");
}
async function noteTelegramUserIdHelp(prompter) {
	await prompter.note([
		`1) DM your bot, then read from.id in \`${formatCliCommand("openclaw logs --follow")}\` (safest)`,
		"2) Or call https://api.telegram.org/bot<bot_token>/getUpdates and read message.from.id",
		"3) Third-party: DM @userinfobot or @getidsbot",
		`Docs: ${formatDocsLink("/telegram")}`,
		"Website: https://openclaw.ai"
	].join("\n"), "Telegram user id");
}
function normalizeTelegramAllowFromInput(raw) {
	return raw.trim().replace(/^(telegram|tg):/i, "").trim();
}
function parseTelegramAllowFromId(raw) {
	const stripped = normalizeTelegramAllowFromInput(raw);
	return /^\d+$/.test(stripped) ? stripped : null;
}
async function promptTelegramAllowFrom(params) {
	const { cfg, prompter, accountId } = params;
	const resolved = resolveTelegramAccount({
		cfg,
		accountId
	});
	const existingAllowFrom = resolved.config.allowFrom ?? [];
	await noteTelegramUserIdHelp(prompter);
	const token = params.tokenOverride?.trim() || resolved.token;
	if (!token) await prompter.note("Telegram token missing; username lookup is unavailable.", "Telegram");
	return patchChannelConfigForAccount({
		cfg,
		channel: "telegram",
		accountId,
		patch: {
			dmPolicy: "allowlist",
			allowFrom: await promptResolvedAllowFrom({
				prompter,
				existing: existingAllowFrom,
				token,
				message: "Telegram allowFrom (numeric sender id; @username resolves to id)",
				placeholder: "@username",
				label: "Telegram allowlist",
				parseInputs: splitOnboardingEntries,
				parseId: parseTelegramAllowFromId,
				invalidWithoutTokenNote: "Telegram token missing; use numeric sender ids (usernames require a bot token).",
				resolveEntries: async ({ token: tokenValue, entries }) => {
					return await Promise.all(entries.map(async (entry) => {
						const numericId = parseTelegramAllowFromId(entry);
						if (numericId) return {
							input: entry,
							resolved: true,
							id: numericId
						};
						const stripped = normalizeTelegramAllowFromInput(entry);
						if (!stripped) return {
							input: entry,
							resolved: false,
							id: null
						};
						const id = await fetchTelegramChatId({
							token: tokenValue,
							chatId: stripped.startsWith("@") ? stripped : `@${stripped}`
						});
						return {
							input: entry,
							resolved: Boolean(id),
							id
						};
					}));
				}
			})
		}
	});
}
async function promptTelegramAllowFromForAccount(params) {
	const accountId = resolveOnboardingAccountId({
		accountId: params.accountId,
		defaultAccountId: resolveDefaultTelegramAccountId(params.cfg)
	});
	return promptTelegramAllowFrom({
		cfg: params.cfg,
		prompter: params.prompter,
		accountId
	});
}
const telegramOnboardingAdapter = {
	channel,
	getStatus: async ({ cfg }) => {
		const configured = listTelegramAccountIds(cfg).some((accountId) => {
			return inspectTelegramAccount({
				cfg,
				accountId
			}).configured;
		});
		return {
			channel,
			configured,
			statusLines: [`Telegram: ${configured ? "configured" : "needs token"}`],
			selectionHint: configured ? "recommended · configured" : "recommended · newcomer-friendly",
			quickstartScore: configured ? 1 : 10
		};
	},
	configure: async ({ cfg, prompter, options, accountOverrides, shouldPromptAccountIds, forceAllowFrom }) => {
		const defaultTelegramAccountId = resolveDefaultTelegramAccountId(cfg);
		const telegramAccountId = await resolveAccountIdForConfigure({
			cfg,
			prompter,
			label: "Telegram",
			accountOverride: accountOverrides.telegram,
			shouldPromptAccountIds,
			listAccountIds: listTelegramAccountIds,
			defaultAccountId: defaultTelegramAccountId
		});
		let next = cfg;
		const resolvedAccount = resolveTelegramAccount({
			cfg: next,
			accountId: telegramAccountId
		});
		const hasConfigToken = hasConfiguredSecretInput(resolvedAccount.config.botToken) || Boolean(resolvedAccount.config.tokenFile?.trim());
		const accountConfigured = Boolean(resolvedAccount.token) || hasConfigToken;
		const allowEnv = telegramAccountId === DEFAULT_ACCOUNT_ID;
		const canUseEnv = allowEnv && !hasConfigToken && Boolean(process.env.TELEGRAM_BOT_TOKEN?.trim());
		if (!accountConfigured) await noteTelegramTokenHelp(prompter);
		const tokenResult = await promptSingleChannelSecretInput({
			cfg: next,
			prompter,
			providerHint: "telegram",
			credentialLabel: "Telegram bot token",
			secretInputMode: options?.secretInputMode,
			accountConfigured,
			canUseEnv,
			hasConfigToken,
			envPrompt: "TELEGRAM_BOT_TOKEN detected. Use env var?",
			keepPrompt: "Telegram token already configured. Keep it?",
			inputPrompt: "Enter Telegram bot token",
			preferredEnvVar: allowEnv ? "TELEGRAM_BOT_TOKEN" : void 0
		});
		let resolvedTokenForAllowFrom;
		if (tokenResult.action === "use-env") {
			next = applySingleTokenPromptResult({
				cfg: next,
				channel: "telegram",
				accountId: telegramAccountId,
				tokenPatchKey: "botToken",
				tokenResult: {
					useEnv: true,
					token: null
				}
			});
			resolvedTokenForAllowFrom = process.env.TELEGRAM_BOT_TOKEN?.trim() || void 0;
		} else if (tokenResult.action === "set") {
			next = applySingleTokenPromptResult({
				cfg: next,
				channel: "telegram",
				accountId: telegramAccountId,
				tokenPatchKey: "botToken",
				tokenResult: {
					useEnv: false,
					token: tokenResult.value
				}
			});
			resolvedTokenForAllowFrom = tokenResult.resolvedValue;
		}
		if (forceAllowFrom) next = await promptTelegramAllowFrom({
			cfg: next,
			prompter,
			accountId: telegramAccountId,
			tokenOverride: resolvedTokenForAllowFrom
		});
		return {
			cfg: next,
			accountId: telegramAccountId
		};
	},
	dmPolicy: {
		label: "Telegram",
		channel,
		policyKey: "channels.telegram.dmPolicy",
		allowFromKey: "channels.telegram.allowFrom",
		getCurrent: (cfg) => cfg.channels?.telegram?.dmPolicy ?? "pairing",
		setPolicy: (cfg, policy) => setChannelDmPolicyWithAllowFrom({
			cfg,
			channel: "telegram",
			dmPolicy: policy
		}),
		promptAllowFrom: promptTelegramAllowFromForAccount
	},
	disable: (cfg) => setOnboardingChannelEnabled(cfg, channel, false)
};
//#endregion
//#region extensions/telegram/src/normalize.ts
const TELEGRAM_PREFIX_RE = /^(telegram|tg):/i;
function normalizeTelegramTargetBody(raw) {
	const trimmed = raw.trim();
	if (!trimmed) return;
	const prefixStripped = trimmed.replace(TELEGRAM_PREFIX_RE, "").trim();
	if (!prefixStripped) return;
	const parsed = parseTelegramTarget(trimmed);
	const normalizedChatId = normalizeTelegramLookupTarget(parsed.chatId);
	if (!normalizedChatId) return;
	const keepLegacyGroupPrefix = /^group:/i.test(prefixStripped);
	const hasTopicSuffix = /:topic:\d+$/i.test(prefixStripped);
	const chatSegment = keepLegacyGroupPrefix ? `group:${normalizedChatId}` : normalizedChatId;
	if (parsed.messageThreadId == null) return chatSegment;
	return `${chatSegment}${hasTopicSuffix ? `:topic:${parsed.messageThreadId}` : `:${parsed.messageThreadId}`}`;
}
function normalizeTelegramMessagingTarget(raw) {
	const normalizedBody = normalizeTelegramTargetBody(raw);
	if (!normalizedBody) return;
	return `telegram:${normalizedBody}`.toLowerCase();
}
function looksLikeTelegramTargetId(raw) {
	return normalizeTelegramTargetBody(raw) !== void 0;
}
//#endregion
//#region extensions/telegram/src/status-issues.ts
function readTelegramAccountStatus(value) {
	if (!isRecord(value)) return null;
	return {
		accountId: value.accountId,
		enabled: value.enabled,
		configured: value.configured,
		allowUnmentionedGroups: value.allowUnmentionedGroups,
		audit: value.audit
	};
}
function readTelegramGroupMembershipAuditSummary(value) {
	if (!isRecord(value)) return {};
	const unresolvedGroups = typeof value.unresolvedGroups === "number" && Number.isFinite(value.unresolvedGroups) ? value.unresolvedGroups : void 0;
	const hasWildcardUnmentionedGroups = typeof value.hasWildcardUnmentionedGroups === "boolean" ? value.hasWildcardUnmentionedGroups : void 0;
	const groupsRaw = value.groups;
	return {
		unresolvedGroups,
		hasWildcardUnmentionedGroups,
		groups: Array.isArray(groupsRaw) ? groupsRaw.map((entry) => {
			if (!isRecord(entry)) return null;
			const chatId = asString(entry.chatId);
			if (!chatId) return null;
			return {
				chatId,
				ok: typeof entry.ok === "boolean" ? entry.ok : void 0,
				status: asString(entry.status) ?? null,
				error: asString(entry.error) ?? null,
				matchKey: asString(entry.matchKey) ?? void 0,
				matchSource: asString(entry.matchSource) ?? void 0
			};
		}).filter(Boolean) : void 0
	};
}
function collectTelegramStatusIssues(accounts) {
	const issues = [];
	for (const entry of accounts) {
		const account = readTelegramAccountStatus(entry);
		if (!account) continue;
		const accountId = resolveEnabledConfiguredAccountId(account);
		if (!accountId) continue;
		if (account.allowUnmentionedGroups === true) issues.push({
			channel: "telegram",
			accountId,
			kind: "config",
			message: "Config allows unmentioned group messages (requireMention=false). Telegram Bot API privacy mode will block most group messages unless disabled.",
			fix: "In BotFather run /setprivacy → Disable for this bot (then restart the gateway)."
		});
		const audit = readTelegramGroupMembershipAuditSummary(account.audit);
		if (audit.hasWildcardUnmentionedGroups === true) issues.push({
			channel: "telegram",
			accountId,
			kind: "config",
			message: "Telegram groups config uses \"*\" with requireMention=false; membership probing is not possible without explicit group IDs.",
			fix: "Add explicit numeric group ids under channels.telegram.groups (or per-account groups) to enable probing."
		});
		if (audit.unresolvedGroups && audit.unresolvedGroups > 0) issues.push({
			channel: "telegram",
			accountId,
			kind: "config",
			message: `Some configured Telegram groups are not numeric IDs (unresolvedGroups=${audit.unresolvedGroups}). Membership probe can only check numeric group IDs.`,
			fix: "Use numeric chat IDs (e.g. -100...) as keys in channels.telegram.groups for requireMention=false groups."
		});
		for (const group of audit.groups ?? []) {
			if (group.ok === true) continue;
			const status = group.status ? ` status=${group.status}` : "";
			const err = group.error ? `: ${group.error}` : "";
			const baseMessage = `Group ${group.chatId} not reachable by bot.${status}${err}`;
			issues.push({
				channel: "telegram",
				accountId,
				kind: "runtime",
				message: appendMatchMetadata(baseMessage, {
					matchKey: group.matchKey,
					matchSource: group.matchSource
				}),
				fix: "Invite the bot to the group, then DM the bot once (/start) and restart the gateway."
			});
		}
	}
	return issues;
}
//#endregion
export { telegramOnboardingAdapter as i, looksLikeTelegramTargetId as n, normalizeTelegramMessagingTarget as r, collectTelegramStatusIssues as t };
