// .ignored_openclaw/dist/plugin-sdk/whatsapp.js
import { g as normalizeAccountId, h as DEFAULT_ACCOUNT_ID } from "./session-key-CbP51u9x.js";
import { _ as formatPairingApproveHint, a as formatWhatsAppConfigAllowFromEntries, d as resolveWhatsAppConfigDefaultTo, u as resolveWhatsAppConfigAllowFrom } from "./channel-config-helpers-CX9AfjCJ.js";
import { Da as listWhatsAppDirectoryGroupsFromConfig, Fa as createWhatsAppOutboundBase, Ia as resolveWhatsAppGroupIntroHint, La as resolveWhatsAppMentionStripPatterns, Oa as listWhatsAppDirectoryPeersFromConfig, Ra as resolveWhatsAppOutboundTarget, ar as createActionGate, bs as getChatChannelMeta, eo as resolveWhatsAppGroupRequireMention, lr as readStringParam, s as WhatsAppConfigSchema, to as resolveWhatsAppGroupToolPolicy } from "./thread-bindings-DxarhdX6.js";
import "./paths-WR8OhEmw.js";
import "./github-copilot-token-BGYH4ltJ.js";
import "./logger-D-go2oXy.js";
import "./tmp-openclaw-dir-DRPiOszV.js";
import "./subsystem-DOh66yR6.js";
import { d as normalizeE164 } from "./utils-BhiRkVxe.js";
import "./fetch-Cqwcg-Kq.js";
import "./exec-88qB5pUO.js";
import "./thinking-B2k7q5lY.js";
import "./query-expansion-ChYs46vb.js";
import "./logger-Jt8cCPPV.js";
import "./zod-schema.core-GWl_s9xp.js";
import "./redact-CqHkGWow.js";
import "./http-registry-CR-l0QpT.js";
import "./pairing-token-BwgaSu88.js";
import "./ssrf-Dlo7BZU6.js";
import "./fetch-guard-DEyOIg88.js";
import { i as resolveDefaultGroupPolicy, r as resolveAllowlistProviderRuntimeGroupPolicy } from "./runtime-group-policy-CDs_9_tb.js";
import "./registry-N-tffRct.js";
import "./http-body-CLS7yD5S.js";
import { t as emptyPluginConfigSchema } from "./config-schema-D85tZOYm.js";
import { r as migrateBaseNameToDefaultAccount, t as applyAccountNameToChannelSection } from "./setup-helpers-B1M39J3a.js";
import { i as buildChannelConfigSchema } from "./config-schema-BPehl1ON.js";
import { t as resolveWhatsAppHeartbeatRecipients } from "./whatsapp-heartbeat-NYKh4ERY.js";
export { DEFAULT_ACCOUNT_ID, WhatsAppConfigSchema, applyAccountNameToChannelSection, buildChannelConfigSchema, createActionGate, createWhatsAppOutboundBase, emptyPluginConfigSchema, formatPairingApproveHint, formatWhatsAppConfigAllowFromEntries, getChatChannelMeta, listWhatsAppDirectoryGroupsFromConfig, listWhatsAppDirectoryPeersFromConfig, migrateBaseNameToDefaultAccount, normalizeAccountId, normalizeE164, readStringParam, resolveAllowlistProviderRuntimeGroupPolicy, resolveDefaultGroupPolicy, resolveWhatsAppConfigAllowFrom, resolveWhatsAppConfigDefaultTo, resolveWhatsAppGroupIntroHint, resolveWhatsAppGroupRequireMention, resolveWhatsAppGroupToolPolicy, resolveWhatsAppHeartbeatRecipients, resolveWhatsAppMentionStripPatterns, resolveWhatsAppOutboundTarget };
