// openclaw/dist/plugin-sdk/synology-chat.js
import { h as DEFAULT_ACCOUNT_ID } from "./session-key-CbP51u9x.js";
import "./paths-WR8OhEmw.js";
import "./logger-D-go2oXy.js";
import "./tmp-openclaw-dir-DRPiOszV.js";
import "./subsystem-DOh66yR6.js";
import "./utils-BhiRkVxe.js";
import { H as setAccountEnabledInConfigSection } from "./zod-schema.core-GWl_s9xp.js";
import { t as registerPluginHttpRoute } from "./http-registry-CR-l0QpT.js";
import { a as isRequestBodyLimitError, c as requestBodyErrorToText, s as readRequestBodyWithLimit } from "./http-body-CLS7yD5S.js";
import { t as emptyPluginConfigSchema } from "./config-schema-D85tZOYm.js";
import { i as buildChannelConfigSchema } from "./config-schema-BPehl1ON.js";
import { a as createFixedWindowRateLimiter } from "./webhook-memory-guards-BJifApP1.js";
export { DEFAULT_ACCOUNT_ID, buildChannelConfigSchema, createFixedWindowRateLimiter, emptyPluginConfigSchema, isRequestBodyLimitError, readRequestBodyWithLimit, registerPluginHttpRoute, requestBodyErrorToText, setAccountEnabledInConfigSection };
