// openclaw/dist/plugin-sdk/subagent-registry-runtime-CksbXLi_.js
import "./channel-config-helpers-CX9AfjCJ.js";
import { $t as countPendingDescendantRunsExcludingRun, Qt as countPendingDescendantRuns, Zt as countActiveDescendantRuns, en as isSubagentSessionRunActive, in as shouldIgnorePostCompletionAnnounceForSession, nn as replaceSubagentRunAfterSteer, rn as resolveRequesterForChildSession, tn as listSubagentRunsForRequester } from "./thread-bindings-DxarhdX6.js";
import "./paths-WR8OhEmw.js";
import "./github-copilot-token-BGYH4ltJ.js";
import "./logger-D-go2oXy.js";
import "./tmp-openclaw-dir-DRPiOszV.js";
import "./subsystem-DOh66yR6.js";
import "./utils-BhiRkVxe.js";
import "./fetch-Cqwcg-Kq.js";
import "./exec-88qB5pUO.js";
import "./thinking-B2k7q5lY.js";
import "./query-expansion-ChYs46vb.js";
import "./logger-Jt8cCPPV.js";
import "./zod-schema.core-GWl_s9xp.js";
import "./redact-CqHkGWow.js";
import "./http-registry-CR-l0QpT.js";
import "./pairing-token-BwgaSu88.js";
import "./ssrf-Dlo7BZU6.js";
import "./fetch-guard-DEyOIg88.js";
import "./registry-N-tffRct.js";
import "./http-body-CLS7yD5S.js";
export { countActiveDescendantRuns, countPendingDescendantRuns, countPendingDescendantRunsExcludingRun, isSubagentSessionRunActive, listSubagentRunsForRequester, replaceSubagentRunAfterSteer, resolveRequesterForChildSession, shouldIgnorePostCompletionAnnounceForSession };
