// openclaw/dist/plugin-sdk/voice-call.js
import "./channel-config-helpers-CX9AfjCJ.js";
import { _i as resolveOpenAITtsInstructions } from "./thread-bindings-DxarhdX6.js";
import "./paths-WR8OhEmw.js";
import "./github-copilot-token-BGYH4ltJ.js";
import "./logger-D-go2oXy.js";
import "./tmp-openclaw-dir-DRPiOszV.js";
import "./subsystem-DOh66yR6.js";
import { y as sleep } from "./utils-BhiRkVxe.js";
import "./fetch-Cqwcg-Kq.js";
import "./exec-88qB5pUO.js";
import "./thinking-B2k7q5lY.js";
import "./query-expansion-ChYs46vb.js";
import "./logger-Jt8cCPPV.js";
import { A as TtsConfigSchema, M as TtsProviderSchema, j as TtsModeSchema, k as TtsAutoSchema } from "./zod-schema.core-GWl_s9xp.js";
import "./redact-CqHkGWow.js";
import "./http-registry-CR-l0QpT.js";
import "./pairing-token-BwgaSu88.js";
import "./ssrf-Dlo7BZU6.js";
import { t as fetchWithSsrFGuard } from "./fetch-guard-DEyOIg88.js";
import "./registry-N-tffRct.js";
import { a as isRequestBodyLimitError, c as requestBodyErrorToText, s as readRequestBodyWithLimit } from "./http-body-CLS7yD5S.js";
export { TtsAutoSchema, TtsConfigSchema, TtsModeSchema, TtsProviderSchema, fetchWithSsrFGuard, isRequestBodyLimitError, readRequestBodyWithLimit, requestBodyErrorToText, resolveOpenAITtsInstructions, sleep };
